crates/bevy_ecs/src/core/filter.rs

```diff
@@ -31,12 +31,52 @@ pub struct Or<T>(pub T);
 
 /// Query transformer that retrieves components of type `T` that have been mutated since the start of the frame.
 /// Added components do not count as mutated.
+/// This filter is useful as a performance optimization as it means that a system can be made to run only when
+/// it needs to rather than every frame.
+///
+/// Because the ordering of systems can change and this filter is only effective on changes before the query executes
+/// you need to use explicit dependency ordering or ordered stages for these query filters to be useful.
+///
+/// Example:
+/// ```
+/// use bevy::prelude::*;
```

Comment:
you can't use `use bevy::prelude::*;` from inside `bevy_ecs`, you will need to use a more direct path to the struct you want to use... and as `bevy_ecs` is one of the core crate, you don't have access to a lot of stuff without adding dev dependencies.

`Mutated`, `Added` and `Changed` should be reachable as they are defined here, but you can't use `Name` or `Transform`. Simplest solution would probably to create struct in the doc example and have them hidden, see https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example

Reply:
Thanks for the review! Yup, I've got the examples compiling by using `bevy_ecs::` and hiding some structs.